### PR TITLE
Overshoot Verification

### PR DESCRIPTION
**Pull request by thiago-cavalcante**
Added overshoot verification in DSVerifier.

### CODE REVIEW

**Comment by thiago-cavalcante** [comment]
@lucasccordeiro and @lennonchaves Could you please review the code?

File: src/dsverifier.cpp [hunk 1]
```diff
@@ -2319,6 +2297,138 @@ void verify_state_space_settling_time(void)
   }
 }
 
+/*******************************************************************
+ Function: check_overshoot
+
+ Inputs: Matrices A, B, C, D, x0, integer u and p, and double tsr and ts
```
Comment (lucasccordeiro) [inline src/dsverifier.cpp]: Can you please provide the meaning of each variable? You can describe one per line here.

File: src/dsverifier.cpp [hunk 2]
```diff
@@ -2319,6 +2297,138 @@ void verify_state_space_settling_time(void)
...
+
+ Outputs: 1 if it satisfies or 0 if does not
+
+ Purpose: Check if a given overshoot satisfies to a system
+
+ \*******************************************************************/
+int check_overshoot(Eigen::MatrixXd A, Eigen::MatrixXd B,
+        Eigen::MatrixXd C, Eigen::MatrixXd D, Eigen::MatrixXd x0,
```
Comment (lucasccordeiro) [inline src/dsverifier.cpp]: Is the space correct?

File: src/dsverifier.cpp [hunk 3]
```diff
@@ -2319,6 +2297,138 @@ void verify_state_space_settling_time(void)
...
+
+ Purpose: Verify the overshoot property
+
+ \*******************************************************************/
+void verify_state_space_overshoot(void)
+{
+  double u, _POr;
+  int i, j, isStable;
+  dsverifier_messaget dsv_msg;
+//  _controller_fxp = _controller;
```
Comment (lucasccordeiro) [inline src/dsverifier.cpp]: Remove this!

File: src/dsverifier.cpp [hunk 4]
```diff
@@ -3952,6 +4083,16 @@ int main(int argc, char* argv[])
       verify_state_space_settling_time();
       exit(0);
     }
+    else if(dsv_strings.desired_property == "OVERSHOOT")
+    {
+      if(closedloop == true)
```
Comment (lucasccordeiro) [inline src/dsverifier.cpp]: Can you replace this by ``if(closedloop)''?

**Review by lucasccordeiro — CHANGES_REQUESTED** [review]
@thiago-cavalcante: May I ask you to add regression tests to check this overshoot implementation?

File: src/dsverifier.cpp (hunk 4 above)
Comment (thiago-cavalcante) [reply src/dsverifier.cpp]: Done!

File: src/dsverifier.cpp (hunk 3 above)
Comment (thiago-cavalcante) [reply src/dsverifier.cpp]: Done!

File: src/dsverifier.cpp (hunk 2 above)
Comment (thiago-cavalcante) [reply src/dsverifier.cpp]: Done!

File: src/dsverifier.cpp (hunk 1 above)
Comment (thiago-cavalcante) [reply src/dsverifier.cpp]: Done!

File: src/dsverifier.cpp [hunk 5]
```diff
@@ -2429,17 +2477,20 @@ void verify_state_space_overshoot(void)
       x0(i, j) = _controller.x0[i][j];
     }
   }
-
+  std::cout << "Test"<< std::endl;
```
Comment (lucasccordeiro) [inline src/dsverifier.cpp]: Can you please remove these debug messages?

File: src/dsverifier.cpp [hunk 6]
```diff
@@ -2429,17 +2477,20 @@ void verify_state_space_overshoot(void)
...
   isStable = check_state_space_stability();
   if(isStable)
   {
+	  std::cout << "Test2"<< std::endl;
```
Comment (lucasccordeiro) [inline src/dsverifier.cpp]: Can you please remove these debug messages?

File: src/dsverifier.cpp [hunk 7]
```diff
@@ -2338,19 +2379,26 @@ int check_overshoot(Eigen::MatrixXd A, Eigen::MatrixXd B,
...
-  {
-    mp = yss-yp;
-    std::cout << "There is an overshoot of Mp=" << mp << std::endl;
-  }
-  _PO = mp/yss;
+  std::cout << "cplxMag(yss,0)=" << cplxMag(yss,0) << std::endl;
+  std::cout << "cplxMag(yp,0)=" << cplxMag(yp,0) << std::endl;
+  mp = cplxMag(cplxMag(yp,0)-cplxMag(yss,0),0);
+  std::cout << "There is an overshoot of Mp=" << mp << std::endl;
+//  if(yp >= yss)
```
Comment (lucasccordeiro) [inline src/dsverifier.cpp]: Remove this.

**Review by lucasccordeiro — DISMISSED** [review]
Is this PR ready to be reviewed again?

**Comment by thiago-cavalcante** [comment]
@lucasccordeiro No, this PR is not ready to be reviewed because we are improving the algorithm of function peak_output().

File: src/dsverifier.cpp (hunk 5 above)
Comment (thiago-cavalcante) [reply src/dsverifier.cpp]: Done!

File: src/dsverifier.cpp (hunk 7 above)
Comment (thiago-cavalcante) [reply src/dsverifier.cpp]: Done!

**Comment by thiago-cavalcante** [comment]
@lucasccordeiro @lennonchaves  Could you please review this PR again?

File: src/dsverifier.cpp [hunk 8]
```diff
@@ -2319,6 +2325,142 @@ void verify_state_space_settling_time(void)
...
+{
+  double peakV[2];
+  double yss, yp, mp,_PO;
+  int i = 0;
+  yss = y_ss(A, B, C, D, u);
+  peak_output(A, B, C, D, x0, peakV, yss, u);
+  yp = static_cast<double> (peakV[1]);
+  mp = cplxMag(cplxMag(yp,0)-cplxMag(yss,0),0);
+  std::cout << "There is an overshoot of Mp=" << mp << std::endl;
+//  if(yp >= yss)
```
Comment (lucasccordeiro) [inline src/dsverifier.cpp]: May I ask you to remove this code?

**Review by lucasccordeiro — APPROVED** [review]

File: src/dsverifier.cpp (hunk 8 above)
Comment (thiago-cavalcante) [reply src/dsverifier.cpp]: Done!

**Pull request merged**